x/incentive/keeper/payout.go

```diff
@@ -0,0 +1,169 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/vesting"
+	supplyExported "github.com/cosmos/cosmos-sdk/x/supply/exported"
+	"github.com/kava-labs/kava/x/incentive/types"
+	validatorvesting "github.com/kava-labs/kava/x/validator-vesting"
+)
+
+// PayoutClaim sends the timelocked claim coins to the input address
+func (k Keeper) PayoutClaim(ctx sdk.Context, addr sdk.AccAddress, denom string, id uint64) sdk.Error {
+	claim, found := k.GetClaim(ctx, addr, denom, id)
+	if !found {
+		return types.ErrClaimNotFound(k.codespace, addr, denom, id)
+	}
+	claimPeriod, found := k.GetClaimPeriod(ctx, id, denom)
+	if !found {
+		return types.ErrClaimPeriodNotFound(k.codespace, denom, id)
+	}
+	err := k.SendCoinsFromModuleToVestingAccount(ctx, types.IncentiveMacc, addr, sdk.NewCoins(claim.Reward), int64(claimPeriod.TimeLock.Seconds()))
+	if err != nil {
+		return err
+	}
+
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			types.EventTypeClaim,
+			sdk.NewAttribute(types.AttributeKeySender, fmt.Sprintf("%s", addr)),
+		),
+	)
+	return nil
+}
+
+// SendCoinsFromModuleToVestingAccount sends time-locked coins from the input module account to the recipient. If the recipients account is not a vesting account, it is converted to a periodic vesting account and the coins are added to the vesting balance as a vesting period with the input length.
+func (k Keeper) SendCoinsFromModuleToVestingAccount(ctx sdk.Context, senderModule string, recipientAddr sdk.AccAddress, amt sdk.Coins, length int64) sdk.Error {
```

Comment:
How's this for a refactor suggestion? It's basically just splitting the logic already in `SendCoinsFromModuleToVestingAccount` out into separate functions.
```go
func (k Keeper) SendTimeLockedCoinsFromModuleToAccount(...) {
    // This func sends vesting coins to any type of account.
   // Each type needs different logic so do a big typeswitch and call out to specific funcs (these specific funcs are then reusable)
    acc := // get account
    switch acc.(type)
    case ValidtorVesting:
        // return error
    case ModuleAccount:
        // return error
    case PeriodicAccount:
        SendTimeLockedCoinsToPeriodicVestingAccount()
    case BaseAccount:
        SendTimeLockedCoinsToBaseAccount()
    case default:
        // return error
}

func (k Keeper) SendTimeLockedCoinsToBaseAccount(...) {
    // everything currently in the last else section of `SendCoinsFromModuleToVestingAccount`
}

func (k Keeper) SendTimeLockedCoinsToPeriodicVestingAccount( ... ) {
    // get acount
    // use a pure function to calculate the mathy period updates
    // This separates the calculations from state updates making calculations easier to test in isolation, also makes the maths more reusable (although moot, as I'm not sure where else we'd use this exact maths)
    newStartTime, newPeriods := AddPeriodToPeriods(acc.StartTime, acc.Periods, amountToSend, unlockTimestamp)
    // update acc start time, periods, endtime, original vesting, etc
    // store account
}

func AddPeriodToPeriods(startTime, periods, amount, unlockTime) (newStartTime, newPeriods) {
    // basically the code in the middle section of `SendCoinsFromModuleToVestingAccount`
}
```